src/test/java/io/confluent/examples/streams/GenericAvroIntegrationTest.java

```diff
@@ -51,18 +49,23 @@
  */
 public class GenericAvroIntegrationTest {
 
+  // A mocked schema registry for our serdes to use
+  private static final String SCHEMA_REGISTRY_SCOPE = GenericAvroIntegrationTest.class.getName();
+
+
   private static String inputTopic = "inputTopic";
   private static String outputTopic = "outputTopic";
 
   @Test
   public void shouldRoundTripGenericAvroDataThroughKafka() throws Exception {
-    final MockSchemaRegistryClient mockSchemaRegistryClient = new MockSchemaRegistryClient();
 
     final Schema schema = new Schema.Parser().parse(
       getClass().getResourceAsStream("/avro/io/confluent/examples/streams/wikifeed.avsc")
     );
 
-    mockSchemaRegistryClient.register("inputTopic-value", schema);
+    final SchemaRegistryClient schemaRegistryClient = MockSchemaRegistry.getClientForScope(SCHEMA_REGISTRY_SCOPE);
```

Comment:
This is how you get a client for the given scope (you actually just get back the same `MockSchemaRegistryClient` every time you call this method with the same scope).